meta/classes/package_index.bbclass

```diff
@@ -0,0 +1,21 @@
+#
+#   Creates package indices for the IMAGE_PKGTYPE
+#
+
+do_package_index[nostamp] = "1"
+do_package_index[depends] += "${PACKAGEINDEXDEPS}"
+do_package_index[recrdeptask] += 'do_package_write_deb'
+do_package_index[recrdeptask] += 'do_package_write_ipk'
+do_package_index[recrdeptask] += 'do_package_write_rpm'
+
+python do_package_index() {
+    from oe.rootfs import generate_index_files
+    generate_index_files(d)
+}
+
+# Normally the following DUMMYARCHs aren't included but
+# there's no reason not to build an index for them.
```

Comment:
Upstream will probably want a more affirmative comment here, like "We must build package indexes for the SDK dummy architectures, to support scenarios where SDKs are composed from feeds."